routes/categories.js

```diff
@@ -1,12 +1,16 @@
 const express = require('express');
+const { to } = require('await-to-js');
+
 const router = express.Router();
+
 const models = require('../lib/database/mysql/index');
-const { to } = require('await-to-js');
 const utils = require('../data/utils');
+const cache = require('../lib/cache/redis');
+const middlwr_caching = require('../data/middlewares/caching_functions');
 
 
 // Get all categories
-router.get('/', async(req, res) => {
+router.get('/', middlwr_caching.caching_all_catg, async(req, res) => {
```

Comment:
You havn't implemented cache hit and cache miss yet. You are just adding into cache but you are not fetching from cache.